base/applications/mscutils/eventvwr/eventvwr.c

```diff
@@ -4039,28 +4117,66 @@ EventLogPropProc(HWND hDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
                     EndDialog(hDlg, LOWORD(wParam));
                     return (INT_PTR)TRUE;
 
+                case ID_CLEARLOG:
+                    
+                    EventLogFilter = GetSelectedFilter(NULL);
+                    if (EventLogFilter && ClearEvents(EventLogFilter))
+                    {
+                        Refresh(EventLogFilter);
+                        InitPropertiesDlg(hDlg, EventLog);
+                    }
+                    return (INT_PTR)TRUE;
+                
+                case IDC_EDIT_EVENTS_AGE:
+                case IDC_EDIT_MAXLOGSIZE:
+                    if( HIWORD( wParam ) == EN_CHANGE )
+                    {
+                        PropSheet_Changed(GetParent(hDlg), hDlg);
+                    }
+                    return (INT_PTR)TRUE;
+                    
                 case IDC_OVERWRITE_AS_NEEDED:
                 {
                     CheckRadioButton(hDlg, IDC_OVERWRITE_AS_NEEDED, IDC_NO_OVERWRITE, IDC_OVERWRITE_AS_NEEDED);
                     EnableDlgItem(hDlg, IDC_EDIT_EVENTS_AGE, FALSE);
                     EnableDlgItem(hDlg, IDC_UPDOWN_EVENTS_AGE, FALSE);
-                    break;
+                    PropSheet_Changed(GetParent(hDlg), hDlg);
+                    return (INT_PTR)TRUE;
                 }
 
                 case IDC_OVERWRITE_OLDER_THAN:
                 {
                     CheckRadioButton(hDlg, IDC_OVERWRITE_AS_NEEDED, IDC_NO_OVERWRITE, IDC_OVERWRITE_OLDER_THAN);
                     EnableDlgItem(hDlg, IDC_EDIT_EVENTS_AGE, TRUE);
                     EnableDlgItem(hDlg, IDC_UPDOWN_EVENTS_AGE, TRUE);
-                    break;
+                    PropSheet_Changed(GetParent(hDlg), hDlg);
+                    return (INT_PTR)TRUE;
                 }
 
                 case IDC_NO_OVERWRITE:
                 {
                     CheckRadioButton(hDlg, IDC_OVERWRITE_AS_NEEDED, IDC_NO_OVERWRITE, IDC_NO_OVERWRITE);
                     EnableDlgItem(hDlg, IDC_EDIT_EVENTS_AGE, FALSE);
                     EnableDlgItem(hDlg, IDC_UPDOWN_EVENTS_AGE, FALSE);
-                    break;
+                    PropSheet_Changed(GetParent(hDlg), hDlg);
+                    return (INT_PTR)TRUE;
+                }
+                
+                case IDC_RESTOREDEFAULTS:
+                {
+                    LoadStringW(hInst, IDS_APP_TITLE, szTitle, 128);
+                    LoadStringW(hInst, IDS_RESTOREDEFAULTS, szText, 256);
```

Comment:
Consider using `_countof()` to compute the exact character count of the array than hardcoding the value.

Reply:
ok